src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java

```diff
@@ -258,4 +293,141 @@ public static String addQueryParams(String path, Map<String, String> params) {
         return errorParams;
     }
 
+    /**
+     * Wraps JWTUtil.isJWT to simplify test mock.
+     *
+     * @param token The token in question
+     * @return True if the token is a jwt
+     */
+    protected static boolean isJWT(AuthToken token) {
+        return JWTUtil.isJWT(token);
+    }
+
+    /**
+     * Retrieves authToken with jwt from state param as priority.<br>
+     * If no jwt from state param exists, forwards to default cookie/header check.
+     *
+     * @param cookies Request cookies
+     * @param headers Request headers
+     * @param string JWT from client's state param
+     * @return An auth token
+     * @throws ServiceException If there are issues creating the auth token
+     */
+    protected static AuthToken getAuthToken(Cookie[] cookies, Map<String, String> headers, String jwt)
+        throws ServiceException {
+        AuthToken authToken = null;
+        if (!StringUtils.isEmpty(jwt)) {
+            try {
+                jwt = URLDecoder.decode(jwt, OAuth2Constants.ENCODING.getValue());
+                final String salt = JWTUtil.getJWTSalt(jwt);
```

Comment:
Formerly `final String salt = getFromCookie(cookies, ZimbraCookie.JWT_ . . .` but I found this to be error prone.

Looking for feedback here because this may be defeating the original intent of the jwt cookie?